### PR TITLE
Space - Nikki V

### PR DESCRIPTION
**Pull request by vnikki13**
# Task List
Congratulations! You're submitting your assignment!

## Comprehension Questions
Question | Answer
:------------- | :-------------
Describe in your own words what the Model is doing in Rails | Model is the rows in a Table. It contains the name, description, and completed_at information of each task. |
Describe in your own words what the Controller is doing in Rails | The controller contains all the methods associated with a route verb and is responsible for sending back a response. |
Describe in your own words what the View is doing in Rails | View is responsible for showing a readable layout for the user. Does not contain any Ruby business logic. |
Describe an edge-case controller test you wrote | Will redirect to the root page given an invalid id. |
What is the purpose of using strong params? (i.e. the params method in the controller) | Use strong params so you don't have to write out all the params every time you want to input some data into the database.
How are Rails migrations related to Rails models? | Rails migrations are related to the Table itself such as adding a new column or creating a new Table. Models are the data or rows in the Table. |
Describe one area of Rails that are still unclear on | Controller helpers. The documentation for them is not very helpful and I'm confused about some of the arguments. I understand the form one. |

### CODE REVIEW

File: app/views/tasks/index.html.erb [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+        method: :patch,
+        disabled: task.completed_at, 
+        params: [
+          task: {
+            name: task.name, 
+            description: task.description, 
+            completed_at: Date.today.to_formatted_s(:long)
+          }
+        ]
+      %>
```
Comment (kaidamasaki) [inline app/views/tasks/index.html.erb:18]: In the future you should make things like this an action in the controller since this is more complicated logic than we usually like to have in a view.

File: db/migrate/20200430203130_create_tasks.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
+class CreateTasks < ActiveRecord::Migration[6.0]
+  def change
+    create_table :tasks do |t|
+      t.string :name
+      t.string :description
+      t.string :completed_at
```
Comment (kaidamasaki) [inline db/migrate/20200430203130_create_tasks.rb:6]: In the future you should use `t.datetime` as the type for columns like this.

**Comment by kaidamasaki** [comment]
# Task List




## Major Learning Goals/Code Review





| Criteria | yes/no, and optionally any details/lines of code to reference |
| --- | --- |
| At least 6 commits with meaningful commit messages | ✔️
| Routes follow RESTful conventions | ✔️
| Uses named routes (like `_path`) | ✔️
| Creates Models and migrations | ✔️
| Creates styled views | Unstyled views.
| Handles errors like nonexistant tasks | ✔️
| Uses `form_with` to render forms in Rails | ✔️

## Functional Requirements/Manual Testing

| Functional Requirement | yes/no |
| --- | --- |
| Successfully handles index & show | ✔️
| index & show tests pass | ✔️
| Successfully handles: New, Create | ✔️
| New, Create tests pass | ✔️
| Successfully handles: Edit, Update | ✔️
| Successfully handles: Destroy, Task Complete | ✔️

## Overall Feedback

| Overall Feedback | Criteria | yes/no |
| --- | --- | --- |
| Green (Meets/Exceeds Standards) | 5+ in Code Review && 5+ in Functional Requirements | ✔️
| Yellow (Approaches Standards) | 3+ in Code Review && 4+ in Functional Requirements, or the instructor judges that this project needs special attention |
| Red (Not at Standard) | 0-2 in Code Review or 0-3 in Functional Reqs, or assignment is breaking/doesn’t run with less than 5 minutes of debugging, or the instructor judges that this project needs special attention |





## Code Style Bonus Awards



Was the code particularly impressive in code style for any of these reasons (or more...?)

| Quality | Yes? |
| --- | --- |
| Perfect Indentation | ✅
| Elegant/Clever | ✅
| Descriptive/Readable | ✅
| Concise | ✅
| Logical/Organized |

**Review by kaidamasaki — COMMENTED** [review]
Good job!  Here are a couple of ways to improve your future assignments. 😃
